### PR TITLE
[UPDATE] socks5 client improvements

### PR DESCRIPTION
**Pull request by twofaktor**
# Description

To be able to run the socks5 client in the background

# Checklist:

- [X] Added "Automating your socks5 client with systemd" section
- [X] Few typo fixes
- [X] Added missing spaces

### CODE REVIEW

File: documentation/docs/src/clients/socks5-client.md [hunk 1]
```diff
@@ -99,19 +102,22 @@ The `--provider` field needs to be filled with the Nym address of a Network Requ
...
 
 However, there are several options for choosing a gateway, if you do not want one that is randomly assigned to your client:
+
 * If you wish to connect to a specific gateway, you can specify this with the `--gateway` flag when running `init`.
 * You can also choose a gateway based on its location relative to your client. This can be done by appending the `--latency-based-selection` flag to your `init` command. This command means that to select a gateway, your client will:
-	* fetch a list of all availiable gateways
+	* fetch a list of all available gateways
 	* send few ping messages to all of them, and measure response times.
-	* create a weighted distribution to randomly choose one, favouring ones with lower latency.
+	* create a weighted distribution to randomly choose one, favoring ones with lower latency.
```
Comment (mfahampshire) [inline documentation/docs/src/clients/socks5-client.md]: this isnt a typo fix but is just changing the spelling from british to american english. Since the docs are written in british english im not going to accept this change.

File: documentation/docs/src/clients/socks5-client.md [hunk 2]
```diff
@@ -134,25 +140,56 @@ The `config.toml` file contains client configuration options, while the two `pem
...
 You can set this via the `--host` flag during either the `init` or `run` commands.
 
 Alternatively, a custom host can be set in the `config.toml` file under the `socket` section. If you do this, remember to restart your client process.
 
-
-
 ### Running the socks5 client
 
-You can run the initalised client by doing this:
+You can run the initialized client by doing this:
```
Comment (mfahampshire) [inline documentation/docs/src/clients/socks5-client.md]: Like the commit above, please change to british english spelling.

File: documentation/docs/src/clients/socks5-client.md [hunk 3]
```diff
@@ -134,25 +140,56 @@ The `config.toml` file contains client configuration options, while the two `pem
...
+systemctl start nym-socks5-client.service
+
+# you can always check your socks5 client has succesfully started with:
+systemctl status nym-socks5-client.service
+```
+
 ## Using your Socks5 Client
 
-After completing the steps above, your local Socks5 Client will be listening on `localhost:1080` ready to proxy traffic to the Network Requester set as the `--provider` when initialising.
+After completing the steps above, your local Socks5 Client will be listening on `localhost:1080` ready to proxy traffic to the Network Requester set as the `--`provider` when initializing.
```
Comment (mfahampshire) [inline documentation/docs/src/clients/socks5-client.md]: looks like there's an extra backtick in `provider` which looks incorrect to me. also please revert the change from british to american english with `initialising`

**Comment by mfahampshire** [comment]
just FYI in the future for docs related PRs please use the `docs` issue template - just so I definitely see them :)

**Comment by twofaktor** [comment]
> just FYI in the future for docs related PRs please use the `docs` issue template - just so I definitely see them :) 

Ok, thanks, I'm not an expert in the English language, I will revert these related changes. About use 'docs' issue, why? This is a pull request with the modifications needed included, not an issue

**Comment by twofaktor** [comment]
I did the changes proposed, could you re-review @mfahampshire?

**Comment by mfahampshire** [comment]
>  About use 'docs' issue, why? This is a pull request with the modifications needed included, not an issue

Because they are automatically tagged with the `documentation` label and I am automatically set as the reviewer, which means I don't miss them in the other PRs for the monorepo.

**Comment by mfahampshire** [comment]
Just so I can make sure to review them asap

File: documentation/docs/src/clients/socks5-client.md [hunk 4]
```diff
@@ -52,13 +52,15 @@ Many existing applications are able to use either the SOCKS4, SOCKS4A, or SOCKS5
 There are 2 pieces of software that work together to send SOCKS traffic through the mixnet: the `nym-socks5-client`, and the `nym-network-requester`.
 
 The `nym-socks5-client` allows you to do the following from your local machine:
-* Take a TCP data stream from a application that can send traffic via SOCKS5.
+
+* Take a TCP data stream from an application that can send traffic via SOCKS5.
```
Comment (mfahampshire) [inline documentation/docs/src/clients/socks5-client.md:56]: i dont understand what this commit is doing - just looks like you've removed then reintroduced a line?

File: README.md [hunk 5]
```diff
@@ -32,7 +32,11 @@ For Typescript components, please see [ts-packages](./ts-packages).
 
 ### Developer chat
 
-You can chat to us in [Keybase](https://keybase.io). Download their chat app, then click **Teams -> Join a team**. Type **nymtech.friends** into the team name and hit **continue**. For general chat, hang out in the **#general** channel. Our development takes places in the **#dev** channel. Node operators should be in the **#node-operators** channel.
+> We used to use Keybase for developer chats, but we have since migrated to Matrix and Discord. We no longer check the old **nymtech.friends** Keybase team.
```
Comment (mfahampshire) [inline README.md]: this is already present in the current master, develop, and release branches. Why is this change here?

File: documentation/docs/src/clients/socks5-client.md [hunk 6]
```diff
@@ -80,7 +82,8 @@ You can check the necessary parameters for the available commands by running:
 ```
 
 ### Initialising a new client instance
-Before you can use the client, you need to initalise a new instance of it, which can be done with the following command:
+
+Before you can use the client, you need to initialise a new instance of it, which can be done with the following command:
```
Comment (mfahampshire) [inline documentation/docs/src/clients/socks5-client.md:86]: see above comment

File: documentation/docs/src/clients/socks5-client.md [hunk 7]
```diff
@@ -134,22 +140,53 @@ The `config.toml` file contains client configuration options, while the two `pem
...
 ### Running the socks5 client
 
-You can run the initalised client by doing this:
+You can run the initialised client by doing this:
 
 ```
 ./nym-socks5-client run --id docs-example
 ```
 
+## Automating your socks5 client with systemd
```
Comment (mfahampshire) [inline documentation/docs/src/clients/socks5-client.md:158]: nice! has this been tested?

**Review by mfahampshire — CHANGES_REQUESTED** [review]

File: documentation/docs/src/clients/socks5-client.md (hunk 7 above)
Comment (twofaktor) [reply documentation/docs/src/clients/socks5-client.md:158]: yes, and running fine!

File: documentation/docs/src/clients/socks5-client.md (hunk 4 above)
Comment (twofaktor) [reply documentation/docs/src/clients/socks5-client.md:56]: no, this is an added missing space, following markdown language rules

check "view file" to see the end results are the same, this is just a code level change:

imagen

File: documentation/docs/src/clients/socks5-client.md (hunk 6 above)
Comment (twofaktor) [reply documentation/docs/src/clients/socks5-client.md:86]: same above comment

File: README.md (hunk 5 above)
Comment (twofaktor) [reply README.md]: I don't remember to be changed this file 😕

File: documentation/docs/src/clients/socks5-client.md (hunk 7 above)
Comment (twofaktor) [reply documentation/docs/src/clients/socks5-client.md:158]: BTW: what do you think about adding `journalctl -fu nym-socks5-client` command to monitor the logs in real-time? this could be useful to use in other components of the network (for example network requesters) also

File: README.md (hunk 5 above)
Comment (twofaktor) [reply README.md]: Do you want that I delete the file of this PR?

File: documentation/docs/src/clients/socks5-client.md (hunk 7 above)
Comment (mfahampshire) [reply documentation/docs/src/clients/socks5-client.md:158]: I'm worried about starting to add stuff like `journalctl` commands because i dont want the docs to get bloated - if anything I want them to get a bit more streamlined.. do you think its necessary?

File: README.md (hunk 5 above)
Comment (mfahampshire) [reply README.md]: yes if you can please remove those commits - it looks like this PR actually has 3 of my commits in it, at the beginning, which have already been committed to the main branches of the repo

File: documentation/docs/src/clients/socks5-client.md (hunk 4 above)
Comment (mfahampshire) [reply documentation/docs/src/clients/socks5-client.md:56]: ok let me take a look

File: documentation/docs/src/clients/socks5-client.md (hunk 7 above)
Comment (twofaktor) [reply documentation/docs/src/clients/socks5-client.md:158]: For me monitoring the logs in real time is important to detect possible issues, but could be a personal preference

**Comment by twofaktor** [comment]
I couldn't delete README.md file of this PR. I create a new PR: https://github.com/nymtech/nym/pull/3546 in the new [nymtech:release/v1.1.21](https://github.com/nymtech/nym/tree/release/v1.1.21) branch
